.github/workflows/test-ui.yml

```diff
@@ -88,6 +100,55 @@ jobs:
           jwtGithubAudience: ${{ vars.CI_VAULT_AUD }}
           secrets: |-
             kv/data/teams/nomad/ui PERCY_TOKEN ;
+      - name: Download all test results
+        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          pattern: test-results-*
+          path: test-results
+
+      - name: Combine test results for comparison
+        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+        # Note: iterator is hardcoded to 4 to match matrix partitions
+        run: |
```

Comment:
Should this script be in the `scripts/` or `ui/` directory rather than embedded in yaml?

Reply:
That's a great idea; separating this out now.